include/swift/AST/DiagnosticsSema.def

```diff
@@ -3526,6 +3543,10 @@ ERROR(unsupported_convention,none,
 ERROR(unreferenced_generic_parameter,none,
       "generic parameter '%0' is not used in function signature", (StringRef))
 
+// Opaque types
+ERROR(unsupported_opaque_type,none,
+      "'opaque' types are only implemented for the declared type of properties or the return type of functions", ())
```

Comment:
What about subscript element types?

Reply:
Should this mention "read-only" for properties and subscripts?